docs/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,174 @@
+        - They aren't in the same modules. But they are similar concepts. This
+          helps discoverability and signals to other humans the behavior of the
+          symbols without having to consume the implementation.
+        - over-specifying function names, types, etc.: 
+            - counteracts re-use and discoverability
+            - creates churn when the implementation changes, because then the name needs to change (it doesn't really matter that "parsing" is being done here)
+- Use common names unless there is a strong, conscious reason to use
+  a variant/uncommon name.
+    - `Info` is more common that `Details`.
+    - "Get" is more common than "Retrieve"
+        - "Fetch" might be used to indicate that special work is being done vs
+          some other existing (or potential) "Get" counterpart in the same
+          module. Ask: If the function is _not_ doing special work then why is
+          it named "Fetch" instead of "Get"?
+- Over-specifying names:
+    - Counteracts re-use and discoverability
+    - Creates churn when the implementation changes, because then the name
+      needs to change (it doesn't really matter that "parsing" is being done
+      here).
+        - Or worse: the name can't/didn't change, now the function has an
+          overspecified and misleading name.
+
+## Project guidelines
+
+This section documents answers to common questions, decisions, UX and project
+consistency, questions of style and code structure, and anything else that
+cannot be enforced by a `lint` build-task.
+
+- Telemetry: "active" vs "passive"
+  - Active (`passive:false`) metrics are those intended to appear in DAU count.
+- Icons:
+  - Where possible, use IDE-specific standard icons (e.g. standard VSCode
+    standard icons in VSCode, and Cloud9 standard icons in Cloud9). The typical
+    (maintainable) way to do this is to use _named_ icons (what VSCode calls
+    [codicons](https://microsoft.github.io/vscode-codicons/)) as opposed to
+    icons shipped with the Toolkit build and referenced by _path_.
+  - For cases where icons must be statically defined (package.json), if Cloud9
+    does not support the VSCode standard icon, use the Cloud9 icon.
+- Changelog guidelines
+  - Prefer active voice: "You can do X" instead of "X can be done"
+- Avoid unnecessary use of `lodash` (which we may remove in the future).
+  Functions such as `forEach()`, `map()`, `filter()`, and many others are
+  already available from ES6/typescript.
+- Do not habitually define concepts one-per-file. Related classes, interfaces,
+  and symbols should live in the same module unless there is an explicit,
+  conscious motivation for separating them.
+- What is a "unit test"? https://www.youtube.com/watch?v=EZ05e7EMOLM
+  - Unit is not "a single class", it's a piece of the software with a clear
+    boundary. It might be a whole microservice, or a chunk of a monolith that
+    is internally consistent.
+- User-facing messages (and log messages):
```

Comment:
Make a note that log and error messages shouldn't be localized but other user-facing messages should